### PR TITLE
Labs 1465/bundler demo code patch

### PR DESCRIPTION
**Pull request by 108356037**
Modify demo env to Sepolia, and directly interact with pre-deployed accounts.
I've modified the `NonStandardAccount.sol` so  anyone can send tx to it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# ETH Taipei 2023 Account Abstraction Workshop
+\# ETH Taipei 2023 Account Abstraction Workshop
```
Comment (NIC619) [inline README.md]:
```suggestion
# ETH Taipei 2023 Account Abstraction Workshop
```

File: README.md [hunk 2]
```diff
@@ -71,67 +71,133 @@ Running 2 tests for test/SignatureAccount.t.sol:SignatureAccountTest
...
+# }
+```
+
+### Deploy a standard 4337 Account on Sepolia
+Try modifying `contracts/bundler/NonStandardAccount.sol` then deploy the account and play with it. You may get some Sepolia eth on https://sepolia-faucet.pk910.de or https://sepoliafaucet.com/
+
+Hint:
+
+1. Comment out codes that will likely cause failure at Bundler's validation stage.  
+2. (Optional) The `_validateSignature` function of `NonStandardAccount.sol` isn't properly implemented yet, it doesn't do any signature check. Add a verification logic in the function so only wallet owner can use the account.
```
Comment (NIC619) [inline README.md]: No need to implement this here unless we are tasking them with implementing a standard account with knowledge from prior tasks.

File: README.md [hunk 3]
```diff
@@ -71,67 +71,133 @@ Running 2 tests for test/SignatureAccount.t.sol:SignatureAccountTest
...
+---
+
+## Bundler Demo
+
+For this demo, we will interact with two pre-deployed 4337 accounts on Sepolia testnet:
 
-### Generate UserOp payload for bundler
-*(prerequisite: environment needs python3 installed to run below script)*
+1. `0x6137A181E3657A5dfd4Ca97C5bB1d50B3AAdb127`
+2. `0xa729a76caadb6fdcD9c198cEd19b5D4e54bA0485`
```
Comment (NIC619) [inline README.md]:
```suggestion
1. `0x6137A181E3657A5dfd4Ca97C5bB1d50B3AAdb127` (Account accessing BANNED OPCODE)
2. `0xa729a76caadb6fdcD9c198cEd19b5D4e54bA0485` (Account accessing invalid Storage Slot)
```

File: README.md [hunk 4]
```diff
@@ -71,67 +71,133 @@ Running 2 tests for test/SignatureAccount.t.sol:SignatureAccountTest
...
-Write down the deployed account address at this step, we will need it when generating userOp. 
 
+---
+
+## Bundler Demo
+
+For this demo, we will interact with two pre-deployed 4337 accounts on Sepolia testnet:
 
-### Generate UserOp payload for bundler
-*(prerequisite: environment needs python3 installed to run below script)*
```
Comment (pilagod) [inline README.md]: Q: Don't need python anymore?

File: contracts/bundler/NonStandardAccount.sol [hunk 5]
```diff
@@ -62,26 +62,32 @@ contract NonStandardAccount is BaseAccount {
...
-        bytes32 userOpHash
+        bytes32 /*userOpHash*/
     ) internal virtual override returns (uint256 validationData) {
         // Should fail
         uint256 lastBalance = address(this).balance;
         emit bundlerTestCall(address(this), lastBalance);
 
         // Should fail
-        uint256 shoudlNotPassCall = IERC20(0x87224F6D41DF6044ddd30a87bBdEeBc8c8CAc4f0).balanceOf(address(entryPoint()));
+        uint256 shoudlNotPassCall = IERC20(0xf531B8F309Be94191af87605CfBf600D71C2cFe0).balanceOf(address(entryPoint()));
```
Comment (NIC619) [inline contracts/bundler/NonStandardAccount.sol]: Turn the hardcoded address into a constant, easier to read and maintain.

File: contracts/bundler/NonStandardAccount.sol [hunk 6]
```diff
@@ -62,26 +62,32 @@ contract NonStandardAccount is BaseAccount {
...
         uint256 selfStorageNonceCall = this.nonce();
         emit bundlerTestCall(address(this), selfStorageNonceCall);
 
-        bytes32 hash = userOpHash.toEthSignedMessageHash();
-        if (owner != hash.recover(userOp.signature)) return SIG_VALIDATION_FAILED;
+        // To prevent bad actors transferring all the balance
+        (, uint256 value, ) = abi.decode(userOp.callData[4:], (address, uint256, bytes));
+        if (value > 2 wei) {
+            revert("This account is for demo purpose, please don't transfer large funds out from this account...");
+        }
```
Comment (NIC619) [inline contracts/bundler/NonStandardAccount.sol]: This check should happen in `execute`

File: contracts/bundler/NonStandardAccount.sol [hunk 7]
```diff
@@ -62,26 +62,32 @@ contract NonStandardAccount is BaseAccount {
...
+
+        // Returns 0 anyway so everyone can pass signature validation
+        // (Optional) TODO: Add some verification logic here so only wallet owner can use this account
```
Comment (NIC619) [inline contracts/bundler/NonStandardAccount.sol]: No need for this

File: README.md (hunk 1 above)
Comment (pilagod) [inline README.md]:
```suggestion
# ETH Taipei 2023 Account Abstraction Workshop
```
Accident? 🧐

**Review by pilagod — DISMISSED** [review]
Others LGTM

File: script/bundler/DeployAccount.s.sol [hunk 8]
```diff
@@ -13,6 +13,8 @@ contract DeployAccount is Script {
     // Use an account that is under your control, we will need this account's private key to sign userOperation.
     address ownerAccount = vm.envAddress("ACCOUNT_OWNER_ADDR");
 
+    // Deploys contract NonStandardAccount at contracts/bundler/NonStandardAccount.sol
+    // then automatically deposits 0.01 ether to the account from deployer
```
Comment (NIC619) [inline script/bundler/DeployAccount.s.sol]: nit: the ether is deposited to Entrypoint, not the account
```suggestion
    // then deposits 0.01 ether for the account from deployer
```

**Comment by NIC619** [comment]
This `NonStandardAccount` account is for demonstration of BANNED OPCODE and invalid Storage Slot, other stuffs added to the account are only distractions. Remove the import of `BaseAccount` and modify from account example in previous tasks. This helps the participants to focus on the diff between each task, instead of getting a completely different account with many added features all of a sudden.

File: README.md (hunk 1 above)
Comment (108356037) [reply README.md]: Forgot to run `npm run format` 0.0

File: README.md (hunk 4 above)
Comment (108356037) [reply README.md]: Still need, I added the comment back!

**Comment by 108356037** [comment]
Fixed all codes suggested by Nic, `NonStandardAccount` is now more simple and just inherits the `IAccount` interface. The deploy script is also fixed to only deploy `SimpleAccontFactory` & `SimpleAccount` using official sample.

**Review by NIC619 — DISMISSED** [review]

File: contracts/bundler/NonStandardAccount.sol [hunk 9]
```diff
@@ -1,99 +1,59 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.17;
 
-import "oz/utils/cryptography/ECDSA.sol";
 import "oz/token/ERC20/IERC20.sol";
-import { BaseAccount } from "aa/core/BaseAccount.sol";
 import { UserOperation } from "aa/interfaces/UserOperation.sol";
-import { IEntryPoint } from "aa/interfaces/IEntryPoint.sol";
+import { SignatureAccount } from "../SignatureAccount.sol";
```
Comment (pilagod) [inline contracts/bundler/NonStandardAccount.sol]:
```suggestion
```
nit: Unused

File: script/bundler/DeploySimpleAccount.s.sol [hunk 10]
```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+/* solhint-disable avoid-low-level-calls */
+/* solhint-disable no-inline-assembly */
+/* solhint-disable reason-string */
```
Comment (pilagod) [inline script/bundler/DeploySimpleAccount.s.sol]:
```suggestion
/* solhint-disable avoid-low-level-calls no-inline-assembly reason-string */
```

Comment (108356037) [reply script/bundler/DeploySimpleAccount.s.sol]: Fixed!

**Review by pilagod — APPROVED** [review]

**Pull request merged**